Automattic/vip
# [WIP] Separate commands implementation from specific CLI args parser.

## Description

This is the first step of the first step of removal of yargs in favor of commander. 

The general idea is to decouple commands from the specific arguments parser.

The implementation consists of two pieces:

1. BaseCommand - it's the base class from which the rest commands are derived from. 
2. CommandRegistry - a simple singleton that stores all registered commands and invokes them.

### Missing pieces

- [ ] Argument validation - where do we do it? How do we handle positional vs named
- [ ] Missing the allowed arguments definition. How does data shape look for it?
- [ ] Proper TS definitions


## Pull request checklist


## New release checklist


## Steps to Test

Outline the steps to test and verify the PR here.

Example:

1. Check out PR.
1. Run `npm run build`
1. Run `./dist/lib/command.js`
1. Verify example output is printed.
